src/rdiff-backup-delete

```diff
@@ -60,6 +64,71 @@ def _str(value):
         return value.decode('utf-8', errors='replace')
 
 
+# Check if zcat is available.
+GZIP = spawn.find_executable('gzip')
+
+
+class _wrap_close:
+    """
+    Helper for _open() -- a proxy for a file whose close waits for the process.
+    """
+
+    def __init__(self, stream, proc):
+        self._stream = stream
+        self._proc = proc
+
+    def close(self):
+        if self._proc.stdin:
+            self._proc.stdin.close()
+        returncode = self._proc.wait()
+        if self._proc.stdout:
+            self._proc.stdout.close()
+        if returncode == 0:
+            return None
+        return returncode
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+
+    def __getattr__(self, name):
+        return getattr(self._stream, name)
+
+    def __iter__(self):
+        return iter(self._stream)
+
+
+def _open(fn, mode):
+    """
+    Wrapp to open a file with or without compression using zcat/gzip or
```

Comment:
Wrap (one p)

Reply:
And why do you talk again about zcat?

Reply:
I first started by using zcat ang gzip, but finally figure out how to use gzip for both operation. And I didn't review my comments.